exec/rpc/src/main/java/org/apache/drill/exec/rpc/ProtobufLengthDecoder.java

```diff
@@ -82,15 +79,7 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
         } else {
           // need to make buffer copy, otherwise netty will try to refill this buffer if we move the readerIndex forward...
           // TODO: Can we avoid this copy?
-          ByteBuf outBuf;
-          try {
-            outBuf = allocator.buffer(length);
-          } catch (OutOfMemoryException e) {
-            logger.warn("Failure allocating buffer on incoming stream due to memory limits.  Current Allocation: {}.", allocator.getAllocatedMemory());
-            in.resetReaderIndex();
-            outOfMemoryHandler.handle();
-            return;
-          }
+          ByteBuf outBuf = allocator.buffer(length);
```

Comment:
I will look into that. So far, when I was trying to reproduce the issue through Sqlline I would get an OutOfMemory error in a different place (earlier ?) of the RPC layer.